src/write_file.cpp

```diff
@@ -225,16 +226,21 @@ SEXP buildMatrixMixed(CharacterVector v,
         if(!notNAElements[ri]){
           datetmp[ri] = NA_REAL; //IF TRUE, TRUE else FALSE
         }else{
-          // dt_str = as<std::string>(m(ri,i));
           dt_str = m(ri,i);
+
           try{
-            datetmp[ri] = Rcpp::Date(atoi(dt_str.substr(5,2).c_str()), atoi(dt_str.substr(8,2).c_str()), atoi(dt_str.substr(0,4).c_str()) );
-          }catch(...) {
+            // Some values are incorrectly detected as dates. This regex determines if they are numerics.
+            // If so, they are converted to Dates. 
+            Rcpp::Function cTD("convertToDate");
```

Comment:
should be moved into the if condition or  out of the loop to avoid being called unnecessarily